### PR TITLE
Remove dependency to specific calico and cilium versions.

### PR DESCRIPTION
**Pull request by axel7born**
**How to categorize this PR?**

/area networking
/kind technical-debt
/platform openstack

**What this PR does / why we need it**:
This PR removes the dependency to specific calico and cilium versions.
After each api change of a network extension, the new version needed to be vendored here and a new release has to be made in order to pass admission.
By parsing the network into a general map instead into an object of type `NetworkConfig` we can get rid of this dependency. 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy developer
Remove dependency to specific calico and cilium versions.
```

### CODE REVIEW

**Review by kon-angelo — APPROVED** [review]
/lgtm

File: pkg/admission/mutator/shoot.go [hunk 1]
```diff
@@ -89,97 +93,57 @@ func (s *shoot) Mutate(_ context.Context, newObj, oldObj client.Object) error {
...
 
-			shoot.Spec.Networking.ProviderConfig = &runtime.RawExtension{
-				Object: networkConfig,
-			}
+		modifiedJSON, err := json.Marshal(networkConfig)
+		if err != nil {
+			return err
+		}
+		shoot.Spec.Networking.ProviderConfig = &runtime.RawExtension{
+			Raw: modifiedJSON,
```
Comment (kon-angelo) [inline pkg/admission/mutator/shoot.go:134]: Do I remember correctly that the networking extension doesn't care at all about the `TypeMeta` information of the provided JSON ?

File: pkg/admission/mutator/shoot.go [hunk 2]
```diff
@@ -44,6 +40,14 @@ type shoot struct {
 	decoder runtime.Decoder
 }
 
+const (
+	overlayKey         = "overlay"
+	enabledKey         = "enabled"
+	createPodRoutesKey = "createPodRoutes"
+	calicoReleaseName  = "calico"
+	ciliumReleaseName  = "cilium"
+)
```
Comment (kon-angelo) [inline pkg/admission/mutator/shoot.go:49]: Would it look cleaner if it was a :

```

Comment (axel7born) [reply pkg/admission/mutator/shoot.go:49]: Wouldn't we have then the same problem as before, that the config won't parse if there is a new field in the nertwork config?

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**